src/brogue/Items.c

```diff
@@ -6491,6 +6519,10 @@ void apply(item *theItem, boolean recordCommands) {
         autoIdentify(theItem);
     }
 
+    theItem->lastUsed[2] = theItem->lastUsed[1];
+    theItem->lastUsed[1] = theItem->lastUsed[0];
+    theItem->lastUsed[0] = rogue.playerTurnNumber;
```

Comment:
`printSeed()` (the `~` key) shows `playerTurnNumber`.  So when the player presses `~` periodically and subtracts, they're working in player turns.

I think our goal is to show them if this difference would be greater or smaller than the expected number of turns to gain a charge.

If the expected number of turns to regain a charge is displayed in player turns, then I think this all hangs together, and in particular we want to record `rogue.playerTurnNumber` in the `lastUsed` fields.

If the expected number of turns to regain a charge is in _dungeon_ turns, then this gets more confusing.  But I suspect if it is showing dungeon turns, it would be a bug anyway; the player is either looking at `~` or counting key strokes.

Does anyone know what the displayed expected number of turns to regain a charge is measured in?  @pender maybe?

Reply:
I'm pretty sure staves charge in absolute turns. rechargeItemsIncrementally gets called in playerTurnEnded in the environment update loop, which inside the do-while-paralysed loop for the repeating part of the player turns. If there are incorrect measurements of time in player turn count then I think that should be changed. (Player speed is another factor, I'm not sure how that affects either count.)

Reply:
I'll try to set up a pathelogical scenario in wizard mode & see which count seems accurate.

Reply:
You're right.

I temporarily disabled stochastic recharging (so staffs would recharge exactly at their estimated rate), gave wizard mode a +15 haste charm and changed the firebolt staff to 20/20.  If we pop haste and zap the staff, we need to wait 49 turns rather than 25 (or 24) to get a charge.

I'll try it with the absolute turn number and see how that behaves.

Reply:
Hm.  `absoluteTurnNumber` suffers from the same issue: one needs to wait 49 hasted turns, and see 49 turns in the item description, before getting a recharge in the scenario above.

Reply:
Ok, given that the existing recharge time tooltip is about dungeon time, I think we should give this new tooltip in absolute turns - as we've seen they're not the same but at least they're closer than player turns. We'll open another issue for time ambiguities.

Reply:
Switching to `absoluteTurnNumber` sounds fine, but it's tempting to try and fix the timing issue before merge.  The bugged scenario I set up was extreme, but it really only needed a big haste charm to manifest.  Still, if pulling this thread unravels the sweater, it might be better to do that in a separate bug.

Reply:
I think let's merge this now, I think timing changes could be a big change